<launch>
  <arg name="model" default="burger" doc="model type [burger, waffle, waffle_pi]"/>
  <arg name="mode" default="centralized" doc="control type [individual, centralized]"/>

  <include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="world_name" value="$(find mrs_project)/work/ros/worlds/city.world"/>
    <arg name="paused" value="true"/>
    <arg name="use_sim_time" value="true"/>
    <arg name="gui" value="true"/>
    <arg name="headless" value="false"/>
    <arg name="debug" value="false"/>
  </include>

  <group ns="turtlebot_1">
    <include file="$(find mrs_project)/work/ros/launch/one_bot_$(arg mode).launch">
      <arg name="model" value="$(arg model)"/>
      <arg name="name" value="turtlebot_1"/>
      <arg name="role" value="police"/>
      <arg name="x_pos" value="7.5"/>
      <arg name="y_pos" value="4.5"/>
      <arg name="z_pos" value="0.0"/>
    </include>
  </group>

  <group ns="turtlebot_2">
    <include file="$(find mrs_project)/work/ros/launch/one_bot_$(arg mode).launch">
      <arg name="model" value="$(arg model)"/>
      <arg name="name" value="turtlebot_2"/>
      <arg name="role" value="police"/>
      <arg name="x_pos" value="7.5"/>
      <arg name="y_pos" value="1.5"/>
      <arg name="z_pos" value="0.0"/>
    </include>
  </group>

  <group ns="turtlebot_3">
    <include file="$(find mrs_project)/work/ros/launch/one_bot_$(arg mode).launch">
      <arg name="model" value="$(arg model)"/>
      <arg name="name" value="turtlebot_3"/>
      <arg name="role" value="police"/>
      <arg name="x_pos" value="7.5"/>
      <arg name="y_pos" value="-1.5"/>
      <arg name="z_pos" value="0.0"/>
    </include>
  </group>

  <group ns="turtlebot_4">
    <include file="$(find mrs_project)/work/ros/launch/one_bot_$(arg mode).launch">
      <arg name="model" value="$(arg model)"/>
      <arg name="name" value="turtlebot_4"/>
      <arg name="role" value="police"/>
      <arg name="x_pos" value="7.5"/>
      <arg name="y_pos" value="-4.5"/>
      <arg name="z_pos" value="0.0"/>
    </include>
  </group>

  <group ns="turtlebot_5">
    <include file="$(find mrs_project)/work/ros/launch/one_bot_$(arg mode).launch">
      <arg name="model" value="$(arg model)"/>
      <arg name="name" value="turtlebot_5"/>
      <arg name="role" value="baddie"/>
      <arg name="x_pos" value="-4.5"/>
      <arg name="y_pos" value="7.5"/>
      <arg name="z_pos" value="0.0"/>
    </include>
  </group>

  <group ns="turtlebot_6">
    <include file="$(find mrs_project)/work/ros/launch/one_bot_$(arg mode).launch">
      <arg name="model" value="$(arg model)"/>
      <arg name="name" value="turtlebot_6"/>
      <arg name="role" value="baddie"/>
      <arg name="x_pos" value="-4.5"/>
      <arg name="y_pos" value="-7.5"/>
      <arg name="z_pos" value="0.0"/>
    </include>
  </group>
</launch>
